apis/externalsecrets/v1beta1/secretstore_vault_types.go

```diff
@@ -124,7 +124,15 @@ type VaultAppRole struct {
 
 	// RoleID configured in the App Role authentication backend when setting
 	// up the authentication backend in Vault.
-	RoleID string `json:"roleId"`
+	//+optional
+	RoleID string `json:"roleId,omitempty"`
+
+	// Reference to a key in a Secret that contains the App Role ID used
+	// to authenticate with Vault.
+	// The `key` field must be specified and denotes which entry within the Secret
+	// resource is used as the app role id.
+	//+optional
+	RoleRef esmeta.SecretKeySelector `json:"roleRef,omitempty"`
```

Comment:
This would be better used if it was a pointer:
```suggestion
	RoleRef *esmeta.SecretKeySelector `json:"roleRef,omitempty"`
```

Reply:
@gusfcarvalho Sure I can make that change. Any reason there would be a difference between `RoleRef` and `SecretRef`?

Reply:
Imo, the pointer way is better for optional fields. secretRef is kinda mandatory in the sense that it contains the actual secret to connect to vault. While RoleRef is not because you can provide an inline role Id

Reply:
@gusfcarvalho Ok, sure 👍 Let me know what you think of the updates